make/autoconf/flags-ldflags.m4

```diff
@@ -46,11 +46,15 @@ AC_DEFUN([FLAGS_SETUP_LDFLAGS_HELPER],
 [
   # Setup basic LDFLAGS
   if test "x$TOOLCHAIN_TYPE" = xgcc; then
-    # Add -z,defs, to forbid undefined symbols in object files.
-    # add -z,relro (mark relocations read only) for all libs
+    # Add -z defs, to forbid undefined symbols in object files.
+    if test "x$OPENJDK_TARGET_OS" != xbsd; then
```

Comment:
Do we know what fails on FreeBSD?

Reply:
@emaste Yes, it's gcc fails with a reference to undefined symbol `environ` linker error. The build completes with clang, though.

Reply:
Oh, I remember running into that `environ` symbol issue long time ago, maybe on Solaris, or when I've been toying around with the BSD port myself. I think I tried to get to the bottom of it and figure out what was causing this, but I don't recall what I ended up with. Do you have any more information about what this is, and what would be the proper way to link? Ideally, we would not have to forgo `-z defs`, even on BSD.

Reply:
@magicus The [man page](https://man.freebsd.org/cgi/man.cgi?query=environ&apropos=0&sektion=0&manpath=FreeBSD+14.2-RELEASE+and+Ports&arch=default&format=html) isn't too informative, but from what I've gathered, the symbol should be defined when the process is created and resolved by the dynamic loader when the library is loaded. Didn't follow it too deeply at the time, but I agree it's something to look at before proposing this upstream.

Reply:
It seems `environ` is created by "csu" (whatever that is), and it is apparently a known nuisance when porting to FreeBSD: https://reviews.freebsd.org/D30842. (FWIW, reports on the net suggests NetBSD does not have this problem; I don't know about OpenBSD).

There are indications that something like `-Wl,--ignore-unresolved-symbol -Wl,environ` might work, but it is not clear to me exactly which linkers support this option.

Reply:
So I guess this will be needed. But please add a comment about the environ symbol on BSD to explain why we are removing this (otherwise excellent!) check.